From e294dd092ec2ec7800f24d731af5074c110e83de Mon Sep 17 00:00:00 2001
From: Adrian Muraru <amuraru@adobe.com>
Date: Thu, 30 Oct 2014 23:50:02 +0200
Subject: [PATCH 76/82] HBASE-12386 Replication gets stuck following a transient zookeeper error to remote peer cluster

Reason: Bug
Author: Adrian Muraru
Ref: CDH-23512

Conflicts:
	hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSinkManager.java
---
 .../regionserver/ReplicationSinkManager.java       |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSinkManager.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSinkManager.java
index 839db9b..2c36179 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSinkManager.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSinkManager.java
@@ -107,8 +107,8 @@ public class ReplicationSinkManager {
    * @return a replication sink to replicate to
    */
   public SinkPeer getReplicationSink() throws IOException {
-    if (replicationPeers.getTimestampOfLastChangeToPeer(peerClusterId)
-                                                        > this.lastUpdateToPeers) {
+    if (replicationPeers.getTimestampOfLastChangeToPeer(peerClusterId) > this.lastUpdateToPeers ||
+        sinks.isEmpty()) {
       LOG.info("Current list of sinks is out of date, updating");
       chooseSinks();
     }
-- 
1.7.0.4

